contributors/guide/README.md

```diff
@@ -234,6 +235,18 @@ When reviewing PRs from others [The Gentle Art of Patch Review](http://sage.thes
 
 Note: if your pull request isn't getting enough attention, you can use the [#pr-reviews](https://kubernetes.slack.com/messages/pr-reviews) channel on Slack to get help finding reviewers.
 
+## Contributions best practices
+
+- Do write clear and meaningful Git commit messages.
+- Make sure to include _Github Special Keywords_ which references the issue and automatically closes the issue when PR is merged.
+- Make sure you don't include `@mentions` or `GitHub Keywords` in your PR commit message.
+- When you make a small change to a PR like fixing a typo, any style change, or changing grammar. Make sure you squash your commits so that you dont get a large number of commits for a relatively small change.
```

Comment:
```suggestion
- When you make a PR for small change (such as fixing a typo, style change, or grammar fix), please squash your commits so that we can maintain a cleaner git history.
```